src/extensions/customemotes/CustomEmotesLoader.cpp

```diff
@@ -0,0 +1,253 @@
+#include "CustomEmotesLoader.hpp"
+#include "Application.hpp"
+#include "common/NetworkRequest.hpp"
+#include "common/Outcome.hpp"
+#include "extensions/ExtensionsController.hpp"
+#include "messages/Emote.hpp"
+
+template <>
+inline bool qMapLessThanKey<chatterino::EmoteName>(
+    const chatterino::EmoteName &key1, const chatterino::EmoteName &key2)
+{
+    return key1.string < key2.string;
+}
+
+template <>
+inline bool qMapLessThanKey<chatterino::EmoteNameSet>(
+    const chatterino::EmoteNameSet &key1, const chatterino::EmoteNameSet &key2)
+{
+    return key1.getString() < key2.getString();
+}
+
+namespace chatterino {
+
+namespace {
+
+    Url getEmoteLink(const QJsonObject &urls, const QString &emoteScale)
+    {
+        auto emote = urls.value(emoteScale);
+        if (emote.isUndefined())
+        {
+            return {""};
+        }
+
+        assert(emote.isString());
+
+        return {emote.toString()};
+    }
+    void fillInEmoteData(const QJsonObject &urls, const EmoteName &name,
+                         const QString &tooltip, Emote &emoteData)
+    {
+        auto url1x = getEmoteLink(urls, "1");
+        auto url2x = getEmoteLink(urls, "2");
+        auto url3x = getEmoteLink(urls, "3");
+
+        //, code, tooltip
+        emoteData.name = name;
+        emoteData.images =
+            ImageSet{Image::fromUrl(url1x, 1), Image::fromUrl(url2x, 0.5),
+                     Image::fromUrl(url3x, 0.25)};
+        emoteData.tooltip = {tooltip};
+    }
+}  // namespace
+
+CustomEmotesLoader::CustomEmotesLoader()
+{
+}
+
+void CustomEmotesLoader::loadCustomEmotes()
+{
+    for (const auto &link :
+         getApp()->extensions->customEmotes->items.getVector())
+    {
+        this->load(link);
```

Comment:
Making a separate file for the loading of emotes would be preferred.
See bttvEmotes and ffzEmotes